cmd.cpp

```diff
@@ -1,39 +1,44 @@
 #include "cmd.h"
+#include <iostream>
+#include <fstream>
 using namespace std;
 
 void Dictionary::mkdir(string dirName, string currentFolder) {
 	bool found = false;
-	for (unsigned int i = 0; i < this->system.size(); i++) {
-		if (currentFolder == this->system[i].folder && dirName == this->system[i].subfolder) {
-			found = true;
+	if (this->checkIfNameIsValid(dirName)) cout << "Name cannot contain special characters" << endl;
+	else {
+		for (unsigned int i = 0; i < this->system.size(); i++) {
+			if (currentFolder == this->system[i].folder && dirName == this->system[i].subfolder) {
```

Comment:
Ez nekem nagyon fura. itt a folder helyett nem valami "parent" jellegu dolog kellene, subfolder helyett meg name? a system[i] valami folder/subfolder parokat tarol?

Reply:
Jah, es ha igen, akkor mi van akkor, ha ket kulonbozo helyen is van "alma" konyvtaram?

Reply:
A system egy Pair-eket tároló vektor, és a Pairben folder és subfolder van. 
Lehetséges, hogy legyen két külön helyen is ,,alma" könyvtár.

Reply:
Az oke, es akkor ha az "alma"-ban akarok letrehozni egy "korte"-t, akkor honnet fogja tudni, melyikbe csinalja? MEg ennel bonyolultabb utkozesek is elerhetok nyilvan.

Reply:
A System osztály párokat tárol. Egy mappa nevét és az abszolút elérési utat a mappához (subfolder nevet kicserélhetnénk Path szóra is igazából, de ezzel a névvel kezdtük szóval maradt). Egy CurrentDirectoryval követjük melyik mappában vagyunk. Ez nem a mappa neve, hanem az egész út a mappához így kezelni tudjuk az egyforma neveket. A parancsok mindig a CurrentDirectoryban hajtódnak végre és cd parancs esetén ezt módosítjuk nyilván.

Reply:
Akkor refaktoraljatok a kodot, es nevezzetek at normalis nevekre, amik tenyleg azt jelentik, amit tartalmaznak.